edgelet/iotedge/src/support_bundle.rs

```diff
@@ -47,25 +51,33 @@ where
 
     fn execute(self) -> Self::Future {
         println!("Making support bundle");
-        let result = future::result(self.make_state())
-            .and_then(SupportBundle::write_all_logs)
-            .and_then(SupportBundle::write_edgelet_log_to_file)
-            .and_then(SupportBundle::write_docker_log_to_file)
-            .and_then(SupportBundle::write_check_to_file)
-            .and_then(SupportBundle::write_all_inspects)
-            .and_then(SupportBundle::write_all_network_inspects)
-            .map(|state| {
-                println!(
-                    "Created support bundle at {}",
-                    state
-                        .location
-                        .canonicalize()
-                        .unwrap_or_else(|_| state.location)
-                        .to_string_lossy()
-                )
-            });
-
-        Box::new(result)
+
+        match self.output_location {
+            OutputLocation::File(_) => Box::new(
+                Self::bundle_all(future::result(self.make_file_state())).map(|state| {
+                    let path = PathBuf::from(state.output_location.get_file_location());
+                    println!(
+                        "Created support bundle at {}",
+                        path.canonicalize()
+                            .unwrap_or_else(|_| path)
+                            .to_string_lossy()
```

Comment:
Path implements Display trait. Is this conversion required?

Reply:
`Path` doesn't impl `Display`, but it does have a `display()` method that would be better to use here than `to_string_lossy()` (the result of `Display` never allocates a `String`).